src/pkg/services/m365/api/users.go

```diff
@@ -301,6 +318,62 @@ func (c Users) GetMailInbox(
 	return inbox, nil
 }
 
+func (c Users) GetMailboxInfo(
+	ctx context.Context,
+	userID string,
+) (MailboxInfo, error) {
+	mailFolderFound := true
+	// check whether the user is able to access their inbox.
+	// if they cannot, we can assume they are ineligible for exchange backups.
+	inbx, err := c.GetMailInbox(ctx, userID)
+	if err != nil {
+		if err := EvaluateMailboxError(graph.Stack(ctx, err)); err != nil {
+			logger.CtxErr(ctx, err).Error("getting user's mail folder")
+			return MailboxInfo{}, err
+		}
```

Comment:
These conditionals can be combined for cleaner handling.
```suggestion
	if err := EvaluateMailboxError(err); err != nil {
			logger.CtxErr(ctx, err).Error("getting user's mail folder")
			return MailboxInfo{}, graph.Stack(ctx, err)
		}
```

Reply:
Hey! Changes here are mostly a copy paste from existing code, it's not cleaned up yet. Also just FYI, most of the times I put out draft PRs to let CI run through all the tests & catch any glaring issues. That being said, I appreciate the early review comments :)

Reply:
The latter.